### DISCUSSION TITLE
ContractLogicError: execution reverted

### DISCUSSION THREAD

**Discussion by Brandoneos** [smartcontractkit/full-blockchain-solidity-course-py#1641] [Q&A] [answered]
This wasn't directly in the course but came across while tinkering, and was wondering if I could get some help.
So here in this for loop statement, I am trying to loop through around 50 times, and test values 1-50 for tokenIdCounter. Assume I have a working contract live on the mainnet(contract3 is the instance), and I'm calling the ownerOf(uin256 tokenId) function from most NFT smart contracts. But when I get to a token that hasn't been minted yet, I get the error: 
![githubSC](https://user-images.githubusercontent.com/89663097/184517387-fee7d2ab-2647-459d-b58a-589c6a7e3a15.jpg)
![ts2](https://user-images.githubusercontent.com/89663097/184517429-48424977-16f0-481d-96d5-bcb4dc71418c.jpg)
The smart contract itself returns an error and that error shows up in my terminal, thus stopping my code. The problem is I want to sort of test for that error, and I want to print a message every time that error occurs, but it's currently not possible because it just stops my code when getting to the contract3.functions.ownerOf().call() . Would really appreciate if there is a fix or work around for this. Thanks!

**Comment by Brandoneos** [comment, marked as answer]
Okay I got some help from another person, and simply using a try-except worked. Instead of looking for a specific error like "NameError", just having "except" worked for other errors

**Reply by cromewar** [reply]
Nice catch, could you please share more info about how you solved this?
This might be helpful for other members of the community.

**Reply by Brandoneos** [reply]
![Screenshot_4](https://user-images.githubusercontent.com/89663097/184704269-797c8a70-07b3-4893-99ee-d7aa6bd7c6bc.jpg)

So the "try:" part tries to call the contract's function ownerOf(), but since sometimes the tokenId hasn't been minted yet, so the function call will return an error, if that happens then it goes to the "except:" part to print that an error has occured. What I thought before was there was a specific error type that you had to put after the "except:" part like "ContractLogicError" but it looks like if you don't specify the error it's fine and includes all other errors(probably). Hopefully that explains it.

**Reply by cromewar** [reply]
Thanks!!
